### PR TITLE
Make strings in help command *truly* translatable

### LINKED ISSUE #3818: Help isn't translatable - pot files for strings aren't created
Only thing from `help.py` that gets added to `.po` files is help docstring for `[p]help`.

### PR DESCRIPTION
**Pull request by Dav-Git**
### Type

- [X] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes
This closes #3818

### CODE REVIEW

File: redbot/core/commands/help.py [hunk 1]
```diff
@@ -579,19 +579,20 @@ async def command_not_found(self, ctx, help_for, help_settings: HelpSettings):
             ret = await format_fuzzy_results(ctx, fuzzy_commands, embed=use_embeds)
             if use_embeds:
                 ret.set_author(
-                    name=f"{ctx.me.display_name} {T_('Help Menu')}", icon_url=ctx.me.avatar_url
+                    name=_("{ctx.me.display_name} 'Help Menu'").format(ctx=ctx),
```
Comment (Jackenmen) [inline redbot/core/commands/help.py]: Quotes weren't part of the string her.
```suggestion
                    name=_("{ctx.me.display_name} Help Menu").format(ctx=ctx),
```

File: redbot/core/commands/help.py [hunk 2]
```diff
@@ -579,19 +579,20 @@ async def command_not_found(self, ctx, help_for, help_settings: HelpSettings):
...
             else:
                 await ctx.send(ret)
         elif help_settings.verify_exists:
-            ret = T_("Help topic for *{command_name}* not found.").format(command_name=help_for)
+            ret = _("Help topic for *{command_name}* not found.").format(command_name=help_for)
             if use_embeds:
                 ret = discord.Embed(color=(await ctx.embed_color()), description=ret)
                 ret.set_author(
-                    name=f"{ctx.me.display_name} {T_('Help Menu')}", icon_url=ctx.me.avatar_url
+                    name=f"{ctx.me.display_name} {_('Help Menu')}", icon_url=ctx.me.avatar_url
```
Comment (Jackenmen) [inline redbot/core/commands/help.py]: There's still use of `_()` inside f-string here

**Review by Jackenmen — COMMENTED** [review]
This isn't a review, just two things I noticed when looking at this.

**Comment by Dav-Git** [comment]
When reviewing please pay close attention to Lines 594 - 597. For whatever reason pyling is going ham here but flake8 seems to be ok?

File: redbot/core/commands/help.py [hunk 3]
```diff
@@ -479,7 +479,7 @@ async def format_bot_help(self, ctx: Context, help_settings: HelpSettings):
                 if cog_name:
                     title = f"**__{cog_name}:__**"
                 else:
-                    title = f"**__{T_('No Category')}:__**"
+                    title = _("**__('No Category'):__**")
```
Comment (Jackenmen) [inline redbot/core/commands/help.py]: Parentheses and quotes weren't part of the string.
```suggestion
                    title = _("**__No Category:__**")
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]
I ended up reviewing it...
One issue with a string + you need to replace `_` with something else (`__` is what's usually used when `_` is already taken) on line 193 so that it doesn't end up replacing the variable with translator.

**Comment by mikeshardmind** [comment]
For context, `T_` was used here was to make it visually distinct from markdown, not realizing that redgettext wasn't grabbing this as well (other cogs at the time had used `T_` to denote a deferred translation, so I hadn't even checked)

**Review by Jackenmen — APPROVED** [review]
Looks good to me.

**Pull request merged**